update actix2, crossbeam, log, wasmcloud-provider-core

stevelr opened the pull request:
related to wasmcloud actix2 upgrade

stevelr commented:
I've combined the multiple branches into this one. I mistakenly thought it would be better to split each provider into its own branch and PR, but that made it harder to test. So this branch has them all.

brooksmtownsend reviewed (changes requested):
Thank you for this PR, this is incredibly helpful to mass-upgrade to the latest version of actix. Most of my change requests are just around removing now-unused comments and dropping the `wasmcloud-provider-core` version back to `0.1.0`, as I think that the `0.1.1` change doesn't affect any of these providers. The main reasoning behind this is so that we can properly build/test/release these providers and then release the `wasmcloud` resources afterwards, as of right now the two PRs depend on each other

stevelr commented on redis/Cargo.toml:
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "wasmcloud-redis"
-version = "0.11.1"
+version = "0.11.3"
```
You are right. I should be more careful with using up those positive integers. We don't want to run out!

stevelr commented on fs/Cargo.toml:
```diff
@@ -20,8 +20,8 @@ crate-type = ["cdylib", "rlib"]
 env_logger = "0.7.1"
 wasmcloud-actor-blobstore = "0.1.0"
 wasmcloud-actor-core = "0.2.0"
+wasmcloud-provider-core = { path = "../../wasmcloud/crates/wasmcloud-provider-core", version = "0.1.1" }
```
done

stevelr commented on http-server/Cargo.toml:
```diff
@@ -25,15 +25,10 @@ crate-type = ["cdylib", "rlib"]
-actix-multipart = "0.2.0"
-futures = "0.3.5"
-log = "0.4.8"
+wasmcloud-provider-core = { path="../../wasmcloud/crates/wasmcloud-provider-core", version="0.1.1"}
```
done

stevelr commented on nats-kvcache/Cargo.toml:
```diff
@@ -13,15 +13,14 @@ keywords = [
-[lib]
-crate-type = ["cdylib", "rlib"]
+#[lib]
+#crate-type = ["cdylib", "rlib"]
```
done

stevelr commented on redisgraph/src/conversions.rs:
```diff
@@ -41,8 +38,7 @@ pub(crate) fn redisgraph_column_to_common(
             }
-        }
-        _ => Column::default(),
+        } //_ => Column::default(),
```
done

stevelr commented on telnet/Cargo.toml:
```diff
@@ -21,13 +21,13 @@ static_plugin = []
+crossbeam = "0.8.0"
+crossbeam-channel = "0.5.0"
+serde = { version = "1.0", features = ["derive"] }
+wasmcloud-provider-core = { path = "../../wasmcloud/crates/wasmcloud-provider-core", version = "0.1.1" }
```
done

stevelr commented on http-client/Cargo.toml:
```diff
@@ -20,17 +20,16 @@ static_plugin = []
 serde_json = "1.0.61"
 wasmcloud-actor-http-client = "0.1.0"
 wasmcloud-actor-core = "0.2.0"
+wasmcloud-provider-core = { path = "../../wasmcloud/crates/wasmcloud-provider-core", version="0.1.1" }
```
done

stevelr commented on logging/Cargo.toml:
```diff
@@ -24,6 +24,6 @@ static_plugin = [] # Enable to statically compile this into a host
 log = "0.4.14"
 env_logger = "0.8.2"
-wasmcloud-provider-core = "0.1.0"
+wasmcloud-provider-core = {path="../../wasmcloud/crates/wasmcloud-provider-core", version="0.1.1"}
```
done

stevelr commented on redisgraph/Cargo.toml:
```diff
@@ -22,8 +22,8 @@ static_plugin = []
 wasmcloud-actor-core = "0.2.0"
-wascc-codec = "0.9.0"
-log = "0.4.13"
+wasmcloud-provider-core = { path = "../../wasmcloud/crates/wasmcloud-provider-core", version = "0.1.1" }
```
done

stevelr commented on s3/Cargo.toml:
```diff
@@ -23,18 +23,18 @@ static_plugin = []
+bytes = "1.0"
 wasmcloud-actor-core = "0.2.0"
 wasmcloud-actor-blobstore = "0.1.0"
+wasmcloud-provider-core = { path = "../../wasmcloud/crates/wasmcloud-provider-core", version = "0.1.1" }
```
done

stevelr commented:
@brooksmtownsend  Thank you for the extra eyes- I addressed all your comments in this branch - most fixed in 
[commit in this branch](https://github.com/wasmCloud/capability-providers/pull/51/commits/6c75ec8a9fa580eb3b49fd0bfcb062704a8b099a)

stevelr commented:
@autodidaddict btw, I added two new flags to http-server
     WORKERS=N    to create N worker threads instead of the default: number of logical CPUs
     BIND=ADDR,ADDR,...   comma-separated list of addresses to bind. Each addresses can be IPV4:PORT, IPV6:PORT, or just PORT.  It would be ok to deprecate the PORT flag since BIND does that too.  If no BIND or PORT are provided, the default is the same as before: 0.0.0.0:8080

stevelr reviewed (commented):
Done

stevelr reviewed (commented):
Reviewed.

brooksmtownsend commented on fs/Cargo.toml:
```diff
@@ -20,8 +20,8 @@ crate-type = ["cdylib", "rlib"]
 env_logger = "0.7.1"
 wasmcloud-actor-blobstore = "0.1.0"
 wasmcloud-actor-core = "0.2.0"
+wasmcloud-provider-core = { path = "../../wasmcloud/crates/wasmcloud-provider-core", version = "0.1.0" }
```
This will still unfortunately cause our actions to fail, as the `path` argument is used in local tests rather than the crates.io version. To save some PR comments I'll just leave this on this Cargo.toml, but all of these dependencies will need to be changed to:
```
wasmcloud-provider-core = "0.1.0"
```

brooksmtownsend replied:
Having the `path` argument also affects our Makefile, as the builds for different architectures take place in a docker container which doesn't have the same access to the local filesystem

autodidaddict commented on telnet/src/lib.rs:
```diff
@@ -72,58 +65,40 @@ impl TelnetProvider {
-                .build(),
-        )?)
+    fn health(&self) -> MessageHandlerResult {
+        let bytes = serialize(HealthCheckResponse {
```
This can be converted to just using the `serialize` call without the `Ok` wrapper.

stevelr replied:
fixed in https://github.com/wasmCloud/capability-providers/pull/51/commits/cb89190ff562b02078e9e3424a2bc62db4108fae

autodidaddict reviewed (changes requested):
LGTM after a minor change and @brooksmtownsend minor change request.

brooksmtownsend reviewed (changes requested):
HTTPServer changes look great (and work great), my only request at this point regards the `path` argument that overrides the `wasmcloud-provider-core` version.

stevelr commented on nats-kvcache/Cargo.toml:
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "wasmcloud-nats-kvcache"
-version = "0.5.0"
+version = "0.5.2"
```
You are right. I need to be more sparing with those positive integers. We don't want to run out when we need more later!

autodidaddict commented:
Have we (@brooksmtownsend ) looked into why the nats-kvcache test is failing? Is it passing on local workstations?

brooksmtownsend commented:
@autodidaddict let me clone this locally and take a look

brooksmtownsend commented:
@autodidaddict looks like it fails here: https://github.com/stevelr/capability-providers/blob/actix2-http-server/nats-kvcache/tests/integration.rs#L55

I assume this means that the nats-kvcache provider failed to replicate an event properly

autodidaddict commented:
Try replacing those thread sleeps in the test with tokio delays.

brooksmtownsend commented:
@autodidaddict Because the `handle_call` function now operates by creating a tokio runtime and blocking until the work is completed, using the tokio sleep methods will cause a panic (because the test now needs to be async, and doing that using the `#[tokio::test]` macro spawns a runtime). Essentially tokio runtimes can't be nested, so attempting to call the `handle_call` function now from within a tokio runtime will panic.
I feel I'm missing something obvious... is there a way around this?

autodidaddict commented:
This runtime nesting happens far too often. We know the tests worked back when we were using regular NATS, so let's go back to regular NATS and see if this makes everything happy. I'm not convinced we need to be async in this provider when all of the calls have to be synchronous anyway.

autodidaddict commented:
This LGTM though it appears it needs one more run through `cargo fmt` to pass the check.

brooksmtownsend reviewed (approved):
LGTM
